src/coreclr-debug/activate.ts

```diff
@@ -106,50 +106,72 @@ function showDotnetToolsWarning(message: string): void {
+        return new Promise<void>((resolve, reject) => {
+            let util = new CoreClrDebugUtil(common.getExtensionPath());
+
+            // Check for .debugger folder. Handle if it does not exist.
+            if (!CoreClrDebugUtil.existsSync(util.debugAdapterDir())) {
+                // our install.complete file does not exist yet, meaning we have not completed the installation. Try to figure out what if anything the package manager is doing
+                // the order in which files are dealt with is this:
+                // 1. install.Begin is created
+                // 2. install.Lock is created
+                // 3. install.Begin is deleted
+                // 4. install.complete is created
+    
+                // install.Lock does not exist, need to wait for packages to finish downloading.
+                let installLock = Promise.resolve(false);
+    
+                let debuggerPackage = getRuntimeDependencyPackageWithId("Debugger", this.packageJSON, this.platformInfo, this.extensionPath);
+                if (debuggerPackage && debuggerPackage.installPath)
+                {
+                    installLock = common.installFileExists(debuggerPackage.installPath, common.InstallFileType.Lock);
+                }
+    
+                installLock.then(installLock => {
+                    if (!installLock) {
+                        this.eventStream.post(new DebuggerNotInstalledFailure());
+                        reject(new Error('The C# extension is still downloading packages. Please see progress in the output window below.'));
+                    }
+                    // install.complete does not exist, check dotnetCLI to see if we can complete.
+                    else if (!CoreClrDebugUtil.existsSync(util.installCompleteFilePath())) {
+                        completeDebuggerInstall(this.platformInfo, this.eventStream).then(success => {
+                            if (!success) {
+                                this.eventStream.post(new DebuggerNotInstalledFailure());
+                                reject(new Error('Failed to complete the installation of the C# extension. Please see the error in the output window below.'));
+                            }
+                        });
+                    }
 
-            if (!success) {
-                eventStream.post(new DebuggerNotInstalledFailure());
-                throw new Error('Failed to complete the installation of the C# extension. Please see the error in the output window below.');
+                    resolve();
+                });
             }
-        }
+            else
+            {
+                resolve();
+            }
+        }).then(() => {
+            // debugger has finished installation, kick off our debugger process
+            return new vscode.DebugAdapterExecutable(path.join(common.getExtensionPath(), ".debugger", "vsdbg-ui" + CoreClrDebugUtil.getPlatformExeExtension()));
+        }, error => {
+            throw error;
```

Comment:
>throw error [](start = 12, length = 11)

if you're using this style, I think you probably want to `reject(error)` here, since you have the rejection handler from your new Promise.